src/redux/books/books.js

```diff
@@ -0,0 +1,31 @@
+import Book from '../../components/Book';
+
+// actions
+
+const ADD = 'ADD';
+const REMOVE = 'REMOVE';
+
+// reducer
+
+export default function booksReducer(state = [], action) {
+  switch (action.type) {
+    case ADD:
+      return [...state, action.payload];
```

Comment:
Great job here. the logic is correct. For your question, In order to link to the book react component, you'll have to first wrap your app in index.js with a provided store which you'll learn about in future lessons, then you can use Use Selector to get objects from the provider store.

Reply:
Thanks for taking the time and check!